include/deal.II/lac/block_vector.h

```diff
@@ -458,6 +457,54 @@ void swap (BlockVector<Number> &u,
   u.swap (v);
 }
 
+
+namespace internal
+{
+  namespace LinearOperator
+  {
+    template <typename> class ReinitRangeFactory;
+    template <typename> class ReinitDomainFactory;
+
+    /**
+     * A factory class internally used in linear_operator.h.
+     * Specialization for BlockVector<number>.
+     */
+    template<typename number>
+    class ReinitRangeFactory<BlockVector<number> >
+    {
+    public:
+      template <typename Matrix>
+      std::function<void(BlockVector<number> &, bool)>
+      operator()(const Matrix &matrix)
+      {
+        return [&matrix](BlockVector<number> &v, bool fast)
```

Comment:
Am 13. Apr 2015, 23:32 schrieb Timo Heister <notifications@github.com>:

>> +    template <typename> class ReinitRangeFactory;
>> +    template <typename> class ReinitDomainFactory;
>> +
>> +    /**
>> +     * A factory class internally used in linear_operator.h.
>> +     * Specialization for BlockVector<number>.
>> +     */
>> +    template<typename number>
>> +    class ReinitRangeFactory<BlockVector<number> >
>> +    {
>> +    public:
>> +      template <typename Matrix>
>> +      std::function<void(BlockVector<number> &, bool)>
>> +      operator()(const Matrix &matrix)
>> +      {
>> +        return [&matrix](BlockVector<number> &v, bool fast)
>
> It looks like this code is not guarded with an ifdef to make sure we
> have c++11. Can you please compile your whole PR without it to check?

This is a very good point. Yes, we have to guard. There is an
<code>std::function</code> as a return type that will horribly fail with
C++03.